python/lsst/ip/isr/isrTask.py

```diff
@@ -465,6 +466,9 @@ def run(self, ccdExposure, bias=None, linearizer=None, dark=None, flat=None, def
 
         @return a pipeBase.Struct with field:
          - exposure
+        otherDataRef : `lsst.daf.persistence.butlerSubset.ButlerDataRef`
+            optional dataRef, e.g., needed for DECam crosstalk correction
```

Comment:
Isn't it possible to derive the exposure-based `dataRef` from the CCD-based `dataRef`?
And I believe the `run` method is forbidden from receiving a `dataRef`.

Reply:
You're correct that `run` methods generally shouldn't receive a `dataRef`, as per RFC-352. I've added this special case because I need to extract data from other CCDs within `DecamCrosstalkTask`. I use the `dataRef` to get a Butler and retrieve one or more different exposures with different ccdnums as needed. It's not clear to me how I could move this into a `runDataRef` method as suggested by RFC-352, but I'm open to suggestions.

I also thought I'd preserved old-style docstrings here for consistency (this was before the AHM hack day doc-venture), so I'm impressed you found this briefly-existing numpydoc version :)

Reply:
I think this will work:

```
def getVisitSensorRefs(sensorRef):
    """Return a data reference for all sensors in the visit

    Parameters
    ----------
    sensorRef : `lsst.daf.persistence.ButlerDataRef`
        Data reference for a single sensor.

    Returns
    -------
    visitSensorRefs : `list` of `lsst.daf.persistence.ButlerDataRef`
        Data references for all sensors in the visit.
    """
    butler = sensorRef.getButler()
    visitKeys = butler.getKeys("raw", "visit")
    dataId = sensorRef.dataId.copy()
    for kk in set(dataId.keys).difference(set(visitKeys)):
        dataId.pop(kk)
    visitRef = butler.dataRef("raw", "visit", dataId)
    return [rr for rr in visitRef.subItems("ccd")]
```

Reply:
The established pattern is that `runDataRef` loads everything that's required and passes that to the `run` method. I am concerned that it's naive (because loading the entire focal plane at once for every CCD is not good), but them's the rules. You'll need to get @RobertLuptonTheGood to sign off on your design.

Reply:
Note that the decision on RFC-352 was "We will permit exceptions to that, by means of RFC — anybody who is (re-)designing or implementing a task, and who doesn't feel that run is appropriate, can file an RFC to explain why and get an exemption." We haven't fixed up our existing cases either.

Your suggested `getVisitSensorRefs()` seems to have its own problems of butler dataId manipulation.

Reply:
What's the conclusion here?  I agree with John that there are cases when we might need to pass objects that look a bit like dataRefs around (reference catalogues are one), but I didn't see an RFC to permit it here.

 Why is this in `DecamCrosstalkTask` rather than generalising the cross-talk correction code?

Reply:
Hey folks — just to be clear, the aim behind this PR was for @mrawls to iterate with @SimonKrughoff on the approach she should take here. As of now, that discussion is still ongoing; let's see how it converges. Should any RFCs or other special measures be needed, rest assured that everything will be done by the book.

Reply:
I believe the conclusion is to add a (noop) method called `prepCrosstalk` to the default CrosstalkTask.  Then they will override that in the DECam version to assemble need information.   That's a generic solution that other crosstalk implementations can use.